testsuite/forge/src/main.rs

```diff
@@ -72,6 +77,79 @@ impl PublicUsageTest for FundAccount {
     }
 }
 
+#[derive(Debug)]
+struct TransferCoins;
+
+impl Test for TransferCoins {
+    fn name(&self) -> &'static str {
+        "transfer_coins"
+    }
+}
+
+impl PublicUsageTest for TransferCoins {
+    fn run<'t>(&self, ctx: &mut PublicUsageContext<'t>) -> Result<()> {
+        let mut account = ctx.random_account();
+        let amount = 1000;
+        let currency = Currency::XUS;
+        let client = ctx.client();
+        ctx.fund(
+            currency,
+            account.authentication_key(),
+            amount,
+            ctx.tx_factory(),
+        )?;
+
+        let mut payer = ctx.random_account();
+        let payee = ctx.random_account();
+        let create_payer =
+            account.sign_with_transaction_builder(ctx.tx_factory().create_child_vasp_account(
+                currency,
+                payer.authentication_key(),
+                false,
+                100,
+            ));
+        let create_payee =
+            account.sign_with_transaction_builder(ctx.tx_factory().create_child_vasp_account(
+                currency,
+                payee.authentication_key(),
+                false,
+                0,
+            ));
+        let batch = vec![
+            MethodRequest::submit(&create_payer)?,
+            MethodRequest::submit(&create_payee)?,
+        ];
+        client.batch(batch)?;
+        client.wait_for_signed_transaction(&create_payer, None, None)?;
+        client.wait_for_signed_transaction(&create_payee, None, None)?;
+        let account_view = client.get_account(payer.address())?.into_inner().unwrap();
+        let balance = account_view
+            .balances
+            .iter()
+            .find(|b| b.currency == currency)
+            .unwrap();
+        assert_eq!(balance.amount, 100);
```

Comment:
Maybe we can add a helper function to do this balance checking, since the code is repeated three times.

Reply:
yea, my later change has some checking fns includes balance etc, it will be in next commit